### PR TITLE
boards: silabs: efr32_radio: Remove duplicate usart0 definition

### PR DESCRIPTION
**Pull request by ghost**
In the <board>.dts file, the definition for usart0 was found to be redundant as the same information is already provided in the included .dtsi file. this commit removes the duplicate definition of usart0, resulting in a cleaner and more maintainable device tree configuration.

### CODE REVIEW

**Comment by Arunmanialagarsamy** [comment]
@gmarull @fkokosinski Need you review

**Review by gmarull — APPROVED** [review]

**Comment by Arunmanialagarsamy** [comment]
@fkokosinski Waiting for your approval, please take a review

**Comment by fkokosinski** [comment]
Thanks for your contribution.

> @fkokosinski Waiting for your approval, please take a review

Please keep in mind that I'm no longer maintaining SiLabs platform in Zephyr. For the current maintainers list consult the `MAINTAINERS.yml` file.

**Review by fkokosinski — APPROVED** [review]

**Pull request merged**
